test/v1/controllers/CreateAmendForeignControllerSpec.scala

```diff
@@ -139,19 +125,19 @@ class CreateAmendForeignControllerSpec
     }
   }
 
-  trait Test extends ControllerTest with AuditEventChecking[GenericAuditDetail] {
+  trait Test extends ControllerTest with AuditEventChecking {
 
     val controller = new CreateAmendForeignController(
       authService = mockEnrolmentsAuthService,
       lookupService = mockMtdIdLookupService,
-      parser = mockCreateAmendForeignRequestParser,
+      validatorFactory = mockCreateAmendForeignValidatorFactory,
       service = mockCreateAmendForeignService,
       auditService = mockAuditService,
       cc = cc,
       idGenerator = mockIdGenerator
     )
 
-    protected def callController(): Future[Result] = controller.createAmendForeign(nino, taxYear)(fakePutRequest(requestBodyJson))
+    protected def callController(): Future[Result] = controller.createAmendForeign(validNino, taxYear)(fakePostRequest(requestBodyJson))
```

Comment:
Considering this is the only usage of the method, can we rename back to "fakePutRequest" as the real endpoint uses a put request?

Reply:
Yes - I'll update accordingly